app/models/purl.rb

```diff
@@ -27,19 +25,14 @@ def public_xml
 
   # @return [Hash] the public cocina hash for this Purl object
   def public_cocina
-    @public_cocina ||= JSON.parse(purl_cocina_service.response_body.presence || '{}')
+    @public_cocina ||= JSON.parse(cocina_service_response_body)
```

Comment:
I think if you wanted to avoid re-parsing the JSON, you could have this be `cocina_record.cocina_doc`, which is just the underlying hash parsed from the JSON. In that case you might not even need to memoize, since the call to `cocina_record` is itself memoized.